### ISSUE TITLE
[no-unsafe-call] Triggered on arguments with any type

### ISSUE THREAD

**Issue by gebsh** [typescript-eslint/typescript-eslint#1825]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": "error"
  }
}
```

```TS
let foo: any = 23;
let bar: number = 23;

String(foo); // ERROR: Unsafe call of an any typed value
String(bar); // Works fine
```

**Expected Result**
Pass the test since `foo` is not called but passed as an argument.

**Actual Result**
An error _Unsafe call of an any typed value_ is reported.

**Additional Info**
This works fine when I downgrade `@typescript-eslint/eslint-plugin` to `2.25.0`.

**Versions**
| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.26.0` |
| `@typescript-eslint/parser`        | `2.26.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.12.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Oops, this is a regression accidentally introduced in #1800
